src/PackageManagement.VisualStudio/Runtime/BindingRedirectManager.cs

```diff
@@ -194,7 +194,20 @@ private void Save(string configFileFullPath, XDocument document)
             {
                 document.Save(memoryStream);
                 memoryStream.Seek(0, SeekOrigin.Begin);
-                MSBuildNuGetProjectSystem.AddFile(configFileFullPath, memoryStream);
+
+                // MSBuildNuGetProjectSystem.AddFile() can't handle full path if the app.config
+                // file does not exist in the project yet. This happens when NuGet first creates 
+                // app.config in the project directory. In this case, only the file name is passed.
+                var path = configFileFullPath;
+                var defaultConfigFile = Path.Combine(
+                    MSBuildNuGetProjectSystem.ProjectFullPath,
+                    ConfigurationFile);
+                if (configFileFullPath.Equals(defaultConfigFile, StringComparison.OrdinalIgnoreCase))
+                {
+                    path = ConfigurationFile;
+                }
+
+                MSBuildNuGetProjectSystem.AddFile(path, memoryStream);
```

Comment:
Could you use FileExistsInProject or something to determine if it is being added for the first time? Or, always get the path relative ProjectFullPath before calling AddFile?